system/ticketchallengescache.go

```diff
@@ -0,0 +1,46 @@
+package system
+
+import (
+	"sync"
+	"time"
+)
+
+type ticketChallengesCache struct {
+	sync.Mutex
+	usedSignatures map[string]int64 // [signature]expiry
+}
+
+func newTicketChallengesCache() *ticketChallengesCache {
```

Comment:
I think this is important enough to warrant a comment. It does start an infinite go routine.